homeassistant/components/kostal_plenticore/select.py

```diff
@@ -32,23 +35,39 @@ async def async_setup_entry(
         plenticore,
     )
 
-    async_add_entities(
-        PlenticoreDataSelect(
-            select_data_update_coordinator,
-            entry_id=entry.entry_id,
-            platform_name=entry.title,
-            device_class="kostal_plenticore__battery",
-            module_id=select.module_id,
-            data_id=select.data_id,
-            name=select.name,
-            current_option="None",
-            options=select.options,
-            is_on=select.is_on,
-            device_info=plenticore.device_info,
-            unique_id=f"{entry.entry_id}_{select.module_id}",
-        )
-        for select in SELECT_SETTINGS_DATA
-    )
+    entities = []
+    for select in SELECT_SETTINGS_DATA:
+        if select.module_id in available_settings_data:
+            needed_data_ids = list(
+                filter(lambda data_id: data_id != "None", select.options)
+            )
+            available_data_ids = [
+                setting.id for setting in available_settings_data[select.module_id]
+            ]
+            if all(
+                map(
+                    partial(lambda aid, data_id: data_id in aid, available_data_ids),
```

Comment:
This is hard to read. What is the requirement? We may be able to simplify the expression by comparing two sets instead of iterating over two lists.